lib/EarlyFunctionAnalysis/CFGAnalyzer.cpp

```diff
@@ -1001,8 +1012,34 @@ void CallSummarizer::handleCall(MetaAddress CallerBlock,
 
   Builder.CreateCall(PreCallHook, Args);
 
+  const auto SymbolName = extractFromConstantStringPtr(SymbolNamePointer);
+  const auto &[CalleeSummary, IsTail] = Oracle.getCallSite(Caller,
+                                                           CallerBlock,
+                                                           Callee,
+                                                           SymbolName);
```

Comment:
You should not use the `Oracle` here. Currently this works, because we're cloning the monotone framework state in `Oracle`:

```c++
Oracle.getLocalFunction(EntryAddress).ABIResults = ABIResults
```

But this should go away, it's confusing, we should update the `Oracle` only once at the end and use the state of the monotone framework directly instead.

You need to get access to the state of the monotone framework.
You need to basically get access to the lattice element and query it. See if you can have a reference field to some part o the monotone framework state in the class and initialize it in the constructor of `CallSummarizer`, as we do for `Oracle`.